zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/IndexSchemaValidatorIT.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.schema;
+
+import static java.util.Map.entry;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.json.jackson.JacksonJsonpMapper;
+import co.elastic.clients.transport.ElasticsearchTransport;
+import co.elastic.clients.transport.rest_client.RestClientTransport;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.exporter.NoopExporterConfiguration.ElasticsearchConfig;
+import io.camunda.exporter.NoopExporterConfiguration.IndexSettings;
+import io.camunda.exporter.exceptions.IndexSchemaValidationException;
+import io.camunda.exporter.schema.ElasticsearchEngineClient.MappingSource;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.RestClient;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class IndexSchemaValidatorIT {
```

Comment:
:+1: Nice tests. 
:thought_balloon: Was just wondering if we really have to start ElasticSearch for testing this? Can we not do it with unit tests. Instead of reading the mappings from ElasticSearch, you would have to build it in the tests. That's it, right?

Reply:
right I'll convert it to unit tests.